ExactSolutionEx3.cc

```diff
@@ -0,0 +1,58 @@
+#include "ExactSolutionEx3.hh"
+#include <cmath>
+
+// Example-3, (Taken from numerical experiments of paper AN ADAPTIVE UZAWA FEM FOR THE STOKES PROBLEM, by Morin et al.)
+// This is a problem with smooth solution on square domain (-1,1)*(-1,1)
+
+using std::cos;
+using std::sin;
+using std::pow;
+using std::exp;
+
+template <>
+void ExactSolutionEx3<2>::vector_value(const Point<2> &p, Vector<double> &values) const
+{
+  const double x(p[0]);
+  const double y(p[1]);
+
+  values(0) = 2.*y*cos(pow(x,2)+pow(y,2));
+  values(1) = -2.*x*cos(pow(x,2)+pow(y,2));
+  values(2) = exp(-10.*(pow(x,2)+pow(y,2)));
```

Comment:
I am pretty sure that your exact solution is wrong. The mean value of the pressure is non-zero but in ```StokesProblem::solve``` you subtract the mean value of the pressure to your solution -> the mean value of the pressure of your solution is zero but the mean value of the pressure of the exact solution is non-zero.